src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs

```diff
@@ -1049,5 +1061,44 @@ public void IList_NonGeneric_RemoveAt_ZeroMultipleTimes(int count)
         }
 
         #endregion
+
+        #region Enumerator.Current
+
+        // Enumerator.Current should fail at end after new elements was added
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void IList_NonGeneric_CurrentAtEnd_AfterAdd(int count)
```

Comment:
This code tests non-generic `IEnumerator` interface. In `IList_Generic` similar code tests generic `IEnumerator<T>` interface. `IEnumerator` and `IEnumerator<T>` interfaces can have different behavior in the same class, for example in `List<T>` class. So we should have different code to tests similar interfaces with different behavior.